### ISSUE TITLE
Formatting misfire with anything following item macro call

### ISSUE THREAD

**Issue by udoprog** [rune-rs/rune#527]
See: https://github.com/rune-rs/rune/actions/runs/5084802670/jobs/9137546952?pr=526#step:4:590

**Comment by udoprog** [comment]
It seems like anything following the macro call gets munched, even if it's a use.

**Comment by tgolsson** [comment]
Is there just a missing `\n`? Or more destructive?

**Comment by udoprog** [comment]
It chomps away all newlines (one per fmt call) until there's no separation at all between items.

**Comment by tgolsson** [comment]
I think this invocation needs to add a newline: 

https://github.com/rune-rs/rune/blob/ca44d50651093a436c4c3ac9a936be334f4e561a/crates/rune/src/fmt/printer.rs#L687-L689

So `true, false` as the whitespace args. Super-busy at the moment but might have some time to look more in-depth during the weekend.

**Comment by udoprog** [comment]
No worries. Just thought you'd be interested! Take your time or if you want, I can look into it.

**Comment by tgolsson** [comment]
Hmm, running locally I hit 
```
error: Compile error
  ┌─ scripts/make_function.rn:3:5
  │
3 │     make_function!(inner_fn => { "Hello World Again!" });
  │     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Missing macro `make_function`
```
on latest main.

**Comment by tgolsson** [comment]
Looks to be lookup order, though if I switch the import to the top it fails on the macro evaluation.

**Comment by udoprog** [comment]
Item expansion inside of functions was temporarily disabled until I can figure out how to make it less messy. Try with the current version I just pushed.

I also have a pretty bad cold, so I might be checked out for a few more days.

**Comment by udoprog** [comment]
Closed in #539
